pkg/controller/machine/controller.go

```diff
@@ -168,14 +168,18 @@ func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Resul
 				return reconcile.Result{Requeue: true, RequeueAfter: requeueErr.RequeueAfter}, nil
 			}
 		}
+		return reconcile.Result{}, err
 	}
 	// Machine resource created. Machine does not yet exist.
 	glog.Infof("Reconciling machine object %v triggers idempotent create.", m.ObjectMeta.Name)
 	if err := r.create(m); err != nil {
 		glog.Warningf("unable to create machine %v: %v", name, err)
-		return reconcile.Result{}, err
+		if requeueErr, ok := err.(*controllerError.RequeueAfterError); ok {
+			glog.Infof("Actuator returned requeue-after error: %v", requeueErr)
+			return reconcile.Result{Requeue: true, RequeueAfter: requeueErr.RequeueAfter}, nil
+		}
 	}
-	return reconcile.Result{}, nil
+	return reconcile.Result{}, err
```

Comment:
why return err here?  can this ever not-be nil?  seems like we return on line 158 if `err != nil`, and the other assignments shadow this.

Reply:
It can be not nil:
if` line 175` returns error and is not a `requeueErr` error then 182 must return the error (or nil)

Reply:
Correct me if I am wrong - but doesn't line 175 shadow the err value, so you will never get that assignment:

`if err := r.create(m)`

Reply:
oh yeh, I think you are right! thanks! I'll change it